mlrun/runtimes/nuclio/serving.py

```diff
@@ -709,6 +713,7 @@ def _get_serving_spec(self):
             "track_models": self.spec.track_models,
             "default_content_type": self.spec.default_content_type,
             "model_endpoint_creation_task_name": self.spec.model_endpoint_creation_task_name,
+            "filename": getattr(self.spec, "filename", None),
```

Comment:
Why is this needed? I see in this PR that it's used in tests to run the job locally with custom classes, and it can work with `code_to_function` but how will this work with `project.set_function()` (which is what users are expected to be using now)?